src/tests/multihost/ad/test_access_control.py

```diff
@@ -0,0 +1,506 @@
+        """
+        :title: Set access provider to deny all users
+        :id: 36f82023-3029-4fe4-82b1-7390f01bb5d6
+        :description: Set access_provider to deny This should allow log in of any valid AD-user
+        :steps:
+         1.Set  access_provider to deny in sssd.conf
+         2.Log in as any AD-user
+        :expectedresults:
+         1. SSSD should restart correctly
+         2. Log in should be denied
+        """
+        (aduser, _) = create_aduser_group
+        tools = sssdTools(multihost.client[0], multihost.ad[0])
+        domain_name = tools.get_domain_section_name()
+        multihost.client[0].run_command('realm deny --all', raiseonerr=False)
+        tools.clear_sssd_cache()
+        client_hostname = multihost.client[0].sys_hostname
+        client = pexpect_ssh(client_hostname, f'{aduser}@{domain_name}', 'Secret123', debug=False)
+        try:
+            client.login()
+        except SSHLoginException:
+            pytest.fail(f'{aduser} failed to login')
+        except pexpect.EOF:
+            log_str = multihost.client[0].get_file_contents('/var/log/secure').decode('utf-8')
+            patt = re.compile(r'Access.*denied for user')
+            assert patt.findall(log_str)
+        else:
+            (stdout, _) = client.command(f'id {aduser}@{domain_name}')
+            client.logout()
+
+    @staticmethod
+    def test_dont_fail_auth_with_allow_rules(multihost, create_aduser_group, backupsssdconf):
+        """
+        :title: Set access_provider to permit all users
+        :id: 4e746a3b-8f8e-4884-930f-2ae8ef72e626
+        :description: When simple_allow_groups has one valid and one invalid group as inputs then
+         sssd should ignore invalid group argument. SSSD should allow member-users from valid group
+         to log in
+        :steps:
+         1. Define simple_allow_group with one valid group and one invalid group
+         2. Log in from that valid-groups's member-user
+        :expectedresults:
+         1. SSSD should restart correctly
+         2. Log in should be allowed
+        """
+        (aduser, adgroup) = create_aduser_group
+        tools = sssdTools(multihost.client[0], multihost.ad[0])
+        domain_name = tools.get_domain_section_name()
+        dom_section = f'domain/{domain_name}'
+        sssd_params = {'access_provider': 'simple',
+                       'simple_allow_groups': f'{adgroup}@{domain_name}, non_grp@{domain_name}'}
+        tools.sssd_conf(dom_section, sssd_params, action='add')
+        tools.clear_sssd_cache()
+        client_hostname = multihost.client[0].sys_hostname
+        client = pexpect_ssh(client_hostname, f'{aduser}@{domain_name}', 'Secret123', debug=False)
+        try:
+            client.login()
+        except SSHLoginException:
+            pytest.fail(f'{aduser} failed to login')
+        except pexpect.EOF:
+            log_str = multihost.client[0].get_file_contents('/var/log/secure').decode('utf-8')
```

Comment:
The log_str is not used.
Same as in comment L124 tools.auth_from_client might be used there.

Reply:
Rectified the log_str, 
Please refer my previous reply for similar replacement.